### PR TITLE
Paperwork Prefabs

### PR DESCRIPTION
**Pull request by realforest2001**
Feedback here so I see it faster: https://discord.com/channels/150315577943130112/1233574245418668042/1233574245418668042
# About the pull request
Adds a bunch of forms that can be taken out of filing cabinets to be filled out. Makes use of paperwork much more practical, and as it's using HTML, significantly more customisable.

This PR has focused on Military Police and the Corporate Liaison, but the system is designed to be completely modular.

While I have thoroughly tested my system works the way I want it to, I would appreciate input on quality/efficiency improvement.

# Explain why it's good for the game
QOL, I think it will encourage people to use paperwork a lot more. I particularly want to see Military Police using it more often particularly in regards to Appeals.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

There are a couple inconsistencies in the images below, but these have been corrected since.

image
image

</details>

# Changelog

:cl:
code: Reorganises how paperwork gets image references, and moves them to a new folder.
add: Added prefab paperwork forms for use by Military Police and the Corporate Liaison.
imageadd: Added a dark background often used by staff faxes for easy referencing.
imageadd: Updated WY and USCM logos to higher quality versions.
code: Reorganises the fountain pen into a subtype of obj/item/tool/pen/multicolor to make the functionality usable in multiple pens.
add: Adds a provost pen, a variant of multicolor pen that can switch between Black, White, Orange, Blue, Green and Red. Intended for use in filling out MP forms.
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/game/objects/items/tools/misc_tools.dm
Comment (realforest2001) [inline code/game/objects/items/tools/misc_tools.dm:198]: Assuming this came from a merge conflict.

File: code/modules/reagents/chemistry_machinery/chem_simulator.dm
Comment (realforest2001) [inline code/modules/reagents/chemistry_machinery/chem_simulator.dm:579]:
```suggestion
	report.name = "Simulation result for [D.id]"
```

File: code/modules/reagents/chemistry_machinery/chem_simulator.dm
Comment (realforest2001) [inline code/modules/reagents/chemistry_machinery/chem_simulator.dm:580]:
```suggestion
	report.info += "<center><img src = [asset.get_url_mappings()["logo_wy.png"]]><HR><I><B>Official Company Document</B><BR>Simulated Synthesis Report</I><HR><H2>Result for [D.id]</H2></center>"
```

File: code/modules/paperwork/prefab_papers/wey_yu/liaison/nda_long.dm
Comment (realforest2001) [inline code/modules/paperwork/prefab_papers/wey_yu/liaison/nda_long.dm:123]:
```suggestion
	dat += "This Non-disclosure Agreement is made effective as of <font face=\"Times New Roman\"><i>[time2text(REALTIMEOFDAY, "Day DD Month [GLOB.game_year]")]</i></font> by and between The Company and the United States Colonial Marines <b>(the \"recipient\")</b> of property and information regarding all operational procedures of <span class=\"paper_field\"></span>.</p>"
```

File: code/modules/paperwork/prefab_papers/wey_yu/liaison/liability.dm
Comment (coldironwarrior) [inline code/modules/paperwork/prefab_papers/wey_yu/liaison/liability.dm:123]: hyphenate W-Y

File: code/modules/paperwork/prefab_papers/wey_yu/liaison/liability.dm
Comment (coldironwarrior) [inline code/modules/paperwork/prefab_papers/wey_yu/liaison/liability.dm:129]: ```only for the addressee. If you are not the aforementioned addressee, you should not disseminate```...
add article "the" before addressee. end sentence, start new instruction; comma after "aforementioned addressee", clause stip.

File: code/modules/paperwork/prefab_papers/wey_yu/liaison/preserve_intent.dm
Comment (coldironwarrior) [inline code/modules/paperwork/prefab_papers/wey_yu/liaison/preserve_intent.dm:123]: hyphenate w-y

File: code/modules/paperwork/prefab_papers/wey_yu/liaison/nda_short.dm
Comment (coldironwarrior) [inline code/modules/paperwork/prefab_papers/wey_yu/liaison/nda_short.dm:133]: ```only for the addressee. If you are not the aforementioned addressee, you should not disseminate```...
add article "the" before addressee. end sentence, start new instruction; comma after "aforementioned addressee", clause stip.

File: code/modules/paperwork/prefab_papers/wey_yu/liaison/ops_report.dm
Comment (coldironwarrior) [inline code/modules/paperwork/prefab_papers/wey_yu/liaison/ops_report.dm:132]: ```only for the addressee. If you are not the aforementioned addressee, you should not disseminate```...
add article "the" before addressee. end sentence, start new instruction; comma after "aforementioned addressee", clause stip.

File: code/modules/paperwork/prefab_papers/wey_yu/liaison/preserve_intent.dm
Comment (coldironwarrior) [inline code/modules/paperwork/prefab_papers/wey_yu/liaison/preserve_intent.dm:129]: ```only for the addressee. If you are not the aforementioned addressee, you should not disseminate```...
add article "the" before addressee. end sentence, start new instruction; comma after "aforementioned addressee", clause stip.

**Review by coldironwarrior — COMMENTED** [review]
i have to do it, the bureaucratic grammatical review

will get to NDA file later

File: code/game/objects/items/tools/misc_tools.dm
Comment (realforest2001) [reply code/game/objects/items/tools/misc_tools.dm:198]: Nope, it just moved from lower down

**Comment by deathrobotpunch** [comment]
one thing TG has done is making paperwork prefabs printable by the photocopier which would fit this perfectly

**Comment by deathrobotpunch** [comment]
the new WY logo doesn’t look great

**Review by Nanu308 — COMMENTED** [review]
mapping lgtm

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/modules/paperwork/prefab_papers.dm
Comment (Drulikar) [inline code/modules/paperwork/prefab_papers.dm:34]: This has many that report failures. See any game log where this is TM'd. It also doesn't need to be repeated for every filingcabinet on the map. Rather have this proc ran once to initialize say a GLOB so the checks are ran once. Nothing here appears to be specific to the filing cabinet its being run for.

File: code/modules/clothing/head/head.dm
Comment (Drulikar) [inline code/modules/clothing/head/head.dm:365]: This is an incorrect override which is why the missing icons test is flagging this. `pen_black` for example does not exist in helmet_garb/misc.dmi but `hat_pen_black` and `helmet_pen_black` do. So to automatically prefix the state with `hat_` you need to use `PREFIX_HAT_GARB_OVERRIDE`
```suggestion
	/obj/item/tool/pen/multicolor = PREFIX_HAT_GARB_OVERRIDE,
```

File: code/modules/clothing/head/helmet.dm
Comment (Drulikar) [inline code/modules/clothing/head/helmet.dm:251]: This is an incorrect override which is why the missing icons test is flagging this. `pen_black` for example does not exist in helmet_garb/misc.dmi but `hat_pen_black` and `helmet_pen_black` do. So to automatically prefix the state with `helmet_` you need to use `PREFIX_HELMET_GARB_OVERRIDE`
```suggestion
	/obj/item/tool/pen/multicolor = PREFIX_HELMET_GARB_OVERRIDE,
```

**Review by Drulikar — CHANGES_REQUESTED** [review]

File: code/modules/paperwork/prefab_papers.dm
Comment (realforest2001) [reply code/modules/paperwork/prefab_papers.dm:34]: I should have removed these a long time ago, I haven't needed them since near the start of the PR when I couldn't figure out why things weren't working.

File: code/game/machinery/fax_machine.dm
Comment (realforest2001) [reply code/game/machinery/fax_machine.dm:297]: I have no idea how this got here. It's not meant to be in this PR.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

File: code/modules/paperwork/prefab_papers.dm
Comment (Drulikar) [inline code/modules/paperwork/prefab_papers.dm:49]:
```suggestion
			if(remaining_documents <= 0)
```

File: code/modules/paperwork/prefab_papers.dm
Comment (Drulikar) [inline code/modules/paperwork/prefab_papers.dm:56]:
```suggestion
	if(remaining_documents <= 0)
```

File: code/modules/paperwork/paper.dm
Comment (Drulikar) [inline code/modules/paperwork/paper.dm:372]: This will only eliminate unwritten fields (because there'll be content between the <span>...</span> tags. You need different handling to seal written fields. 
image

File: code/modules/paperwork/paper.dm
Comment (Drulikar) [inline code/modules/paperwork/paper.dm:371]:
```suggestion
	if(tgui_alert(usr, "Are you sure you wish to seal this document? Note: This will prevent the edit of any fields on the paper, excluding the end of the page.", "Confirm", list("Yes", "No"), 20 SECONDS) != "Yes")
		return FALSE
```

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by Drulikar — APPROVED** [review]

**Comment by Drulikar** [comment]
Just needs sprite approval.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by Blundir — APPROVED** [review]

**Pull request merged**
